File: Programmers/Lv1/폰켓몬.cpp
#include <vector>
using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    bool b[200001] = {false};
    for (int i=0; i<nums.size() && answer < nums.size()/2; i++) {
        if (!b[nums[i]]) {
            b[nums[i]] = true;
            answer++;
        }
    }
    return answer;
}

File: Programmers/Lv1/1845_폰켓몬/폰켓몬.js
function solution(nums) {
    var answer = 0;
    let set = new Set();
    const N = nums.length;
    for (let i=0; i<N; i++) {
        set.add(nums[i]);
    }
    answer = set.size > N/2 ? N/2 : set.size;
    return answer; 
}